specification/azurefleet/AzureFleet.Management/fleet.tsp

```diff
@@ -211,6 +228,232 @@ union EvictionPolicy {
+model VMAttributeMinMax {
+  @minValue(0)
+  @doc("Min VMSize from CRS, Min = 0 (uint.MinValue) if not specified.")
+  min?: int32;
+
+  @maxValue(4294967295)
+  @doc("Maz VMSize from CRS, Max = 4294967295 (uint.MaxValue) if not specified.")
+  max?: int32;
+}
+
+@added(Versions.v2024_11_01)
+@doc("VMSizes supported by Azure VMs. Included is a union of Excluded and Required.")
+union VMAttributeSupport {
+  string,
+
+  @doc("All VMSizes having the feature support will be excluded.")
+  Excluded: "Excluded",
+
+  @doc(" VMSizes that have the feature support and that do not have the feature support will be used. Included is a union of Excluded and Required.")
+  Included: "Included",
+
+  @doc("Only the VMSizes having the feature support will be used.")
+  Required: "Required",
+}
+
+@added(Versions.v2024_11_01)
+@doc("""
+  VMCategories defined for Azure VMs.
+  See: https://learn.microsoft.com/en-us/azure/virtual-machines/sizes/overview?tabs=breakdownseries%2Cgeneralsizelist%2Ccomputesizelist%2Cmemorysizelist%2Cstoragesizelist%2Cgpusizelist%2Cfpgasizelist%2Chpcsizelist#general-purpose
+  """)
+union VMCategory {
+  string,
+
+  @doc("""
+    General purpose VM sizes provide balanced CPU-to-memory ratio. Ideal for testing and development, small to medium databases, and low to medium traffic web servers.
+    """)
+  GeneralPurpose: "GeneralPurpose",
+
+  @doc("""
+    Compute optimized VM sizes have a high CPU-to-memory ratio. These sizes are good for medium traffic web servers, network appliances, batch processes, and application servers.
+    """)
+  ComputeOptimized: "ComputeOptimized",
+
+  @doc("""
+    Memory optimized VM sizes offer a high memory-to-CPU ratio that is great for relational database servers, medium to large caches, and in-memory analytics.
+    """)
+  MemoryOptimized: "MemoryOptimized",
+
+  @doc("""
+    Storage optimized virtual machine (VM) sizes offer high disk throughput and IO, and are ideal for Big Data, SQL, NoSQL databases, data warehousing, and large transactional databases. 
+    Examples include Cassandra, MongoDB, Cloudera, and Redis.
+    """)
+  StorageOptimized: "StorageOptimized",
+
+  @doc("""
+    GPU optimized VM sizes are specialized virtual machines available with single, multiple, or fractional GPUs. 
+    These sizes are designed for compute-intensive, graphics-intensive, and visualization workloads.
+    """)
+  GpuAccelerated: "GpuAccelerated",
+
+  @doc("""
+    FPGA optimized VM sizes are specialized virtual machines available with single or multiple FPGA. 
+    These sizes are designed for compute-intensive workloads. This article provides information about the number and type of FPGA, vCPUs, data disks, and NICs. 
+    Storage throughput and network bandwidth are also included for each size in this grouping.
+    """)
+  FpgaAccelerated: "FpgaAccelerated",
+}
+
+@added(Versions.v2024_11_01)
+@doc("Architecture types supported by Azure VMs.")
+union CpuManufacturer {
```

Comment:
Divya has shared a new list of properties. Please use them.

Reply:
Included Divya's updated Spec for VmAttributes.